service/frontend/admin_handler.go

```diff
@@ -238,6 +245,36 @@ func (adh *AdminHandler) Stop() {
 	}
 }
 
+func (adh *AdminHandler) HealthCheck(
```

Comment:
you may want to exclude health check endpoint from rate limiting or make it high priority.

Reply:
good catch.